internal/api/commitment.go

```diff
@@ -850,6 +858,181 @@ func (p *v1Provider) GetCommitmentConversions(w http.ResponseWriter, r *http.Req
 	respondwith.JSON(w, http.StatusOK, map[string]any{"conversions": conversions})
 }
 
+// ConvertCommitment handles POST /v1/domains/{domain_id}/projects/{project_id}/commitments/{commitment_id}/convert
+func (p *v1Provider) ConvertCommitment(w http.ResponseWriter, r *http.Request) {
+	httpapi.IdentifyEndpoint(r, "/v1/domains/:domain_id/projects/:project_id/commitments/:commitment_id/convert")
+	token := p.CheckToken(r)
+	if !token.Require(w, "project:edit") {
+		return
+	}
+	commitmentID := mux.Vars(r)["commitment_id"]
+	if commitmentID == "" {
+		http.Error(w, "no transfer token provided", http.StatusBadRequest)
+		return
+	}
+	dbDomain := p.FindDomainFromRequest(w, r)
+	if dbDomain == nil {
+		return
+	}
+	targetProject := p.FindProjectFromRequest(w, r, dbDomain)
+	if targetProject == nil {
+		return
+	}
+
+	// sourceBehavior
+	var dbCommitment db.ProjectCommitment
+	err := p.DB.SelectOne(&dbCommitment, `SELECT pc.* FROM project_commitments pc WHERE pc.id = $1`, commitmentID)
```

Comment:
We should not allow grabbing commitments from foreign projects:

```suggestion
	err := p.DB.SelectOne(&dbCommitment, findProjectCommitmentByIDQuery, commitmentID, targetProject.ID)
```

Then `targetProject` can be just `project`. The testcases consequently need to be adjusted to not convert between projects (and indeed, there should be a testcase that you cannot convert a foreign commitment without transferring it first).

Reply:
I implemented the query and adjusted the unit tests accordingly.